tests/unit/test_client.py

```diff
@@ -11,29 +11,50 @@
 
 def test_urlopen_proxies(zap, urllib_mock):
     """Check if Zap client passes proxy to urllib call."""
-    urllib_mock.return_value.read.return_value = 'contents'
+    urllib_mock.return_value.text = 'contents'
 
     assert zap.urlopen() == 'contents'
     assert urllib_mock.mock_calls[0][2]['proxies'] == TEST_PROXIES
 
 
 def test_request_response(zap, urllib_mock):
     """Request method should return a python object from parsed output"""
-    urllib_mock.return_value.read.return_value = '{"testkey": "testvalue"}'
+    api_response = {"testkey": "testvalue"}
+    urllib_mock.api_response = api_response
+
+    assert zap._request('http://allizom.org', {'querykey': 'queryvalue'}) == api_response
```

Comment:
Minor comment - _request is expected to be used for API calls rather than 3rd party calls - so 'http://localhost:8080' would be more logical;)